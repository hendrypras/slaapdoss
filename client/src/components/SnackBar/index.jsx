import PropTypes from 'prop-types';
import Snackbar from '@mui/material/Snackbar';

const SnackBar = ({ open, handleClose, message }) => (
  <Snackbar
    data-testid="snackbar"
    autoHideDuration={2000}
    color="red"
    open={open}
    onClose={handleClose}
    message={message}
  />
);
SnackBar.propTypes = {
  open: PropTypes.bool,
  message: PropTypes.string,
  handleClose: PropTypes.func,
};
export default SnackBar;
