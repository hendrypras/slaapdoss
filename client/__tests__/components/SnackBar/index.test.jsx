import { render } from '@utils/testHelper';
import SnackBar from '@components/SnackBar';

let wrapper;
let isOpen = true;
const mockProps = {
  open: isOpen,
  message: 'message',
  onClose: () => {
    isOpen = false;
  },
};
beforeEach(() => {
  wrapper = render(<SnackBar open={mockProps.open} onClose={mockProps.onClose} message={mockProps.message} />);
});

describe('SnackBar', () => {
  test('Should render correctly', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('snackbar')).toBeInTheDocument();
  });
  test('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
