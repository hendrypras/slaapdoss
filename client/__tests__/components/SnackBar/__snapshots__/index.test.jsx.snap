// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnackBar should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft css-cpgvjg-MuiSnackbar-root"
        color="red"
        data-testid="snackbar"
        role="presentation"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root css-1eqdgzv-MuiPaper-root-MuiSnackbarContent-root"
          direction="up"
          role="alert"
          style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
          >
            message
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomLeft css-cpgvjg-MuiSnackbar-root"
      color="red"
      data-testid="snackbar"
      role="presentation"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiSnackbarContent-root css-1eqdgzv-MuiPaper-root-MuiSnackbarContent-root"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiSnackbarContent-message css-1exqwzz-MuiSnackbarContent-message"
        >
          message
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
